source/normalize-arguments.ts

```diff
@@ -105,7 +105,7 @@ export const preNormalizeArguments = (options: any, defaults?: any) => {
 export const normalizeArguments = (url, options, defaults?: any) => {
 	if (is.plainObject(url)) {
 		options = {...url, ...options};
-		url = options.url || {};
+		url = options.url || '';
```

Comment:
This needs testing. `''` if has `baseUrl`, otherwise `{}`.

Reply:
It's an empty string, because merging URLs require two strings (`baseUrl` feature). So you can `customInstance({headers: ...})` without providing the `url` - it will use `baseUrl`. It would throw if the `url` was `{}`.

Reply:
Tested :D